### PR TITLE
made links in system messages clickable

### PR DESCRIPTION
**Pull request by TranRed**
Needed to make sub ticket in sub only chat-message accessible. 
Solution for  #565 

Please review again if there is a smarter way of implementing this. :)

grafik

### CODE REVIEW

File: src/messages/MessageBuilder.cpp [hunk 1]
```diff
@@ -104,8 +104,23 @@ MessageBuilder::MessageBuilder(SystemMessageTag, const QString &text)
     : MessageBuilder()
 {
     this->emplace<TimestampElement>();
-    this->emplace<TextElement>(text, MessageElementFlag::Text,
-                               MessageColor::System);
+
+    // check system message for links
+    // (e.g. needed for sub ticket message in sub only mode)
+    QStringList textFragments = text.split(QRegularExpression("\\s"));
```
Comment (23rd) [inline src/messages/MessageBuilder.cpp]: Imho, `textFragments` can be `const QStringList`.

File: src/messages/MessageBuilder.cpp [hunk 2]
```diff
@@ -104,8 +104,23 @@ MessageBuilder::MessageBuilder(SystemMessageTag, const QString &text)
...
+    for (const auto &word : textFragments)
+    {
+        auto linkString = this->matchLink(word);
```
Comment (23rd) [inline src/messages/MessageBuilder.cpp]: And this one can be `const auto` too.

Comment (TranRed) [reply src/messages/MessageBuilder.cpp]: I don't see any reasons not to

**Comment by TranRed** [comment]
@pajlada since we changed MessageBuilder.cpp in #1045 and #1052 : Will this properly merge?

**Comment by pajlada** [comment]
Looks like it automatically be merged, no problem

**Pull request merged**